### PR TITLE
[pkg/ottl] Add ConvertAttributesToElementsXML Converter

### PR DESCRIPTION
**Pull request by djaglowski**
This adds a converter called `ElementizeAttributesXML`. This serves as one of the granular transformations described in #35281 which will allow users to migrate any arbitrary XML document into a JSON-equivalent state.

Also see #35364

### CODE REVIEW

File: pkg/ottl/ottlfuncs/README.md [hunk 1]
```diff
@@ -600,6 +600,31 @@ Examples:
 - `Duration("333ms")`
 - `Duration("1000000h")`
 
+
+### ElementizeAttributesXML
+
+`ElementizeAttributesXML(target, Optional[xpath])`
+
+The `ElementizeAttributesXML` Converter returns an edited version of an XML string where attributes are converted into child elements.
```
Comment (TylerHelmuth) [inline pkg/ottl/ottlfuncs/README.md:608]: I am worried about the discoverability of this function.  Is `Elementize` a well known term in XML communities? A more verbose, but specific, name would be helpful.

File: pkg/ottl/ottlfuncs/func_elementize_attributes_xml.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+		xPath = "//@*" // All attributes in the document
+	}
+	if err := validateXPath(xPath); err != nil {
+		return nil, err
+	}
+
+	return elementizeAttributesXML(args.Target, xPath), nil
+}
+
+// elementizeAttributesXML returns a `pcommon.String` that is a result of converting all attributes of the
```
Comment (TylerHelmuth) [inline pkg/ottl/ottlfuncs/func_elementize_attributes_xml.go:42]:
```suggestion
// elementizeAttributesXML returns a `string` that is a result of converting all attributes of the
```

File: pkg/ottl/ottlfuncs/README.md (hunk 1 above)
Comment (djaglowski) [reply pkg/ottl/ottlfuncs/README.md:608]: "Element" is a well known term in XML, so I thought "Elementize" conveyed "to make elements", but it's definitely a contrived name. 

Maybe `ConvertAttributesToElementsXML`? I'm open to other suggestions.

Comment (djaglowski) [reply pkg/ottl/ottlfuncs/README.md]: @TylerHelmuth I've updated the name to `ConvertAttributesToElementsXML`. If this looks good to you, I'll make a similar change to https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35364

File: pkg/ottl/ottlfuncs/README.md [hunk 3]
```diff
@@ -544,6 +544,31 @@ Examples:
 
 - `ConvertCase(metric.name, "snake")`
 
+### ConvertAttributesToElementsXML
```
Comment (TylerHelmuth) [inline pkg/ottl/ottlfuncs/README.md:547]: Need to insert the new function in the list at the top of the Converters section.

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**